(WIP) feat: scene editor workflow improvements

misterchedda opened the pull request:
WIP: Scene editing workflow improvements

I've been making some scenes from scratch and implementing some ideas for workflow improvements to make scene editing in general easier. Changes include:

GraphEditor:

- Right click to duplicate node
- Right click to delete connection

![image](https://github.com/user-attachments/assets/bec8ba20-f43b-4e06-923b-738e5ae0e8d7)
![image](https://github.com/user-attachments/assets/5a545730-03d4-4147-88be-a4c47127d6b7)


Tree/Document View:

- Automatic index assignment on creation for actors, performers, props, dialogs
- Dropdown selector UI for actors, performers, props, dialogs, animation resources
- Some minor UI improvements like displaying questNode in scnQuestNode

![image](https://github.com/user-attachments/assets/8ab195fd-4d2c-4f1a-ae3f-afa8b1ba2953)

![image](https://github.com/user-attachments/assets/3912dd21-f695-4d62-970a-1b928fa58534)
![image](https://github.com/user-attachments/assets/7f820c91-ef34-4eac-a4bf-c4a4ae4f2c9b)

manavortex commented on WolvenKit.RED4/Types/Classes/scnActorDef.cs:
```diff
@@ -205,5 +205,47 @@ public scnActorDef()
+		/// Sets up the actor with automatic ID calculation and creates corresponding performer debug symbol
+		/// </summary>
+		/// <param name="sceneResource">The scene resource to add this actor to</param>
+		public void InitializeInScene(scnSceneResource sceneResource)
```
Not sure if this shouldn't go into ClassesExt - @rfuzzo?

manavortex commented on WolvenKit.RED4/Types/Classes/scnPlayerActorDef.cs:
```diff
@@ -170,5 +170,47 @@ public scnPlayerActorDef()
 		}
 
 		partial void PostConstruct();
```
Not sure if this shouldn't go into ClassesExt too

misterchedda commented:
@manavortex I'm going to close this PR, and have included the commits in #2444 where Ive also addressed your review (in this [commit](https://github.com/WolvenKit/WolvenKit/pull/2444/commits/991ce606e89d6db3558b9f332bab7721e338abfc))
